api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java

```diff
@@ -81,7 +83,11 @@
      * Return the properties in this config source
      * @return the map containing the properties in this config source
      */
-    Map<String, String> getProperties();
+    default Map<String, String> getProperties() {
+        Map<String, String> props = new HashMap<>();
+        getPropertyNames().stream().forEach((prop) -> props.put(prop, getValue(prop)));
+        return props;
+    }
 
     /**
      * Gets all property names known to this config source, without evaluating the values.
```

Comment:
I assume "without evaluating the values" should really be "potentially without evaluating the values".

Reply:
updated